cmd/neofs-node/object.go

```diff
@@ -664,20 +661,34 @@ func (o objectSource) Head(ctx context.Context, addr oid.Address) (*objectSDK.Ob
 	return hw.h, err
 }
 
-func (o objectSource) Search(ctx context.Context, cnr cid.ID, filters objectSDK.SearchFilters) ([]oid.ID, error) {
-	var sw searchWriter
-
-	var sPrm searchsvc.Prm
-	sPrm.SetWriter(&sw)
-	sPrm.WithSearchFilters(filters)
-	sPrm.WithContainerID(cnr)
+func (o objectSource) Search(ctx context.Context, cnr cid.ID, filters objectSDK.SearchFilters, childV2 bool) ([]client.SearchResultItem, error) {
```

Comment:
This is not a generic search. It fits the purpose, but it should be named a bit differently then.